lib/sles4sap_publiccloud.pm

```diff
@@ -997,4 +998,42 @@ sub saphanasr_showAttr_version {
     }
 }
 
+=head2 wait_for_zypper
+
+    The function attempts to run 'zypper lr' to check for a lock. If Zypper is locked, it waits for a specified delay before retrying.
+    Returns normally if Zypper is not locked, or dies after a maximum number of retries if Zypper remains locked.
+
+=over 2
+
+=item B<$instance> - The instance object on which the Zypper command is executed. This object must have the run_ssh_command method implemented.
+
+=item B<MAX_RETRIES> - The maximum number of times the function will retry checking if Zypper is locked. Default is 10.
+
+=item B<RETRY_DELAY> - The number of seconds to wait between retries. Default is 20 seconds.
+
+=back
+=cut
+
+sub wait_for_zypper {
+    my ($self, $instance, %args) = @_;
+    my $max_retries = $args{max_retries} // 10;
+    my $retry_delay = $args{retry_delay} // 20;
+    my $retries = 0;
+
+    while ($retries < $max_retries) {
+        my $cmd = 'sudo zypper ref';
```

Comment:
Maybe a variable is not needed and the cmd string can be directly used in `run_ssh_command`

If you prefer to use a variable, probably it does not need to be defined and assigned at each `while` loop, maybe this line can be moved before and outside of it